v0/destinations/snapchat_custom_audience/transform.js

```diff
@@ -0,0 +1,234 @@
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage("Empty/Invalid access token")
+      .build();
+  }
+  return secret.access_token;
+};
+
+/**
+ * Verifies whether the input payload is in right format or not
+ * @param {Object} message
+ * @returns
+ */
+const validatePayload = message => {
+  if (!message.type) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message Type is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message properties is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::listData is not present inside properties. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (message.type.toLowerCase() !== "audiencelist") {
+    throw new ErrorBuilder()
+      .setMessage(
+        `[snapchat_custom_audience]::Message Type ${message.type} not supported.`
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData.add && !message.properties.listData.remove) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Neither 'add' nor 'remove' property is present inside 'listData'. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+};
+
+const generateResponse = (
+  groupedData,
+  schemaType,
+  segmentId,
+  metadata,
+  type
+) => {
+  const payload = { users: [] };
+  const userPayload = { schema: [schemaType], data: groupedData };
+  payload.users.push(userPayload);
+  const response = defaultRequestConfig();
+  if (type === "remove") {
+    response.method = "DELETE";
+    payload.users[0].id = `${segmentId}`;
+  }
+
+  response.endpoint = `${BASE_URL}/segments/${segmentId}/users`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  const accessToken = getAccessToken(metadata);
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json"
+  };
+
+  return response;
+};
+
+const responseBuilder = (metadata, message, { Config }, type) => {
+  const { schema, segmentId } = Config;
+  let data = [[]];
+  let index = 0;
+  let schemaType;
+  let hashedProperty;
+  // Normalizing and hashing ref: https://marketingapi.snapchat.com/docs/#normalizing-hashing
+
+  let userArray;
+  if (type === "add") {
+    userArray = message.properties.listData.add;
+  } else if (type === "remove") {
+    userArray = message.properties.listData.remove;
+  }
+
+  switch (schema) {
+    case "email":
+      schemaType = "EMAIL_SHA256";
+      userArray.forEach(element => {
+        hashedProperty = element?.email;
+        if (!isDefinedAndNotNullAndNotEmpty(hashedProperty)) {
+          return;
```

Comment:
We should throw customError if the required field according to the schema is not available.

Reply:
Done